File: Application/src/main/java/com/dfm/Users/Entities/UserEntity.java
package com.dfm.Users.Entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import jakarta.annotation.Nonnull;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserEntity
{
    @Nonnull
    private String email;
    @Nonnull
    private String password;
    private String firstname;
    private String lastname;
    @Nonnull
    private UserRoleEntity user_role;
}

File: FrontEnd/Projects/Blazor/Users/Entities/UserEntity.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Blazor.Users.Entities;

public class UserEntity
{
    [Required]
    public string email { get; set; }
    [Required]
    public string password { get; set; }
    public string firstname { get; set; }
    public string lastname { get; set; }
    [Required]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public UserRoleEntity user_role { get; set; }
    [JsonIgnore]
    public string fullname => $"{firstname} {lastname}";
}
